internal/dag/extension_processor.go

```diff
@@ -114,6 +118,7 @@ func (p *ExtensionServiceProcessor) buildExtensionService(
 		ClusterTimeoutPolicy: ctp,
 		SNI:                  "",
 		ClientCertificate:    clientCertSecret,
+		UpstreamTLS:          (*UpstreamTLS)(p.UpstreamTLS),
```

Comment:
Huh, I did not realize this was valid Go. Looks like since the API struct and the DAG struct have exactly the same fields, casting is allowed.